org/mozilla/jss/JSSProvider.java

```diff
@@ -3,7 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 package org.mozilla.jss;
 
-public final class JSSProvider extends java.security.Provider {
+public class JSSProvider extends java.security.Provider {
```

Comment:
@edewata We can't encapsulate JSSProviderLoader inside JSSProvider like this:

```java
public final class JSSProvider extends java.security.Provider {
    public final class Loader extends JSSProvider { }
}
``` 

Because it complains about final:

```java
cd /home/ascheel/GitHub/c2/jss/build && /etc/alternatives/java_sdk/bin/javac -classpath /usr/share/java/slf4j/api.jar:/usr/share/java/apache-commons-lang.jar:/usr/share/java/jaxb-api.jar -sourcepath /home/ascheel/GitHub/c2/jss -target 1.8 -source 1.8 -g -d /home/ascheel/GitHub/c2/jss/build/classes/jss -h /home/ascheel/GitHub/c2/jss/build/include/jss/_jni @/home/ascheel/GitHub/c2/jss/build/.targets/java.sources && touch /home/ascheel/GitHub/c2/jss/build/.targets/finished_generate_java
/home/ascheel/GitHub/c2/jss/org/mozilla/jss/JSSProvider.java:346: error: cannot inherit from final JSSProvider
    public final static class JSSProviderLoader extends JSSProvider {
                                                        ^
/home/ascheel/GitHub/c2/jss/org/mozilla/jss/provider/java/security/JSSProviderLoader.java:12: error: cannot inherit from final JSSProvider
public class JSSProviderLoader extends JSSProvider {
                                       ^
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/ascheel/GitHub/c2/jss/org/mozilla/jss/ssl/SSLAlertDescription.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
2 errors
```

My preference would be to leave `JSSProvider` as final somehow. Should I try and add detection logic to `JSSProvider` to see if a CM has been initialized, and if not, call `JSSProviderLoader`'s `init(...)` method? That way we can keep the config-parsing logic separate from the `JSSProvider` itself, and point to `org.mozilla.jss.JSSProvider` as the actual provider in the `java.security` file.